modules/data/src/test/java/io/fluo/webindex/data/IndexIT.java

```diff
@@ -0,0 +1,334 @@
+    config.setInstanceZookeepers(cluster.getZooKeepers() + "/fluo");
+    config.setAccumuloZookeepers(cluster.getZooKeepers());
+    config.setAccumuloTable("data" + tableCounter.getAndIncrement());
+    config.setWorkerThreads(5);
+
+    // create and configure export table
+    exportTable = "export" + tableCounter.getAndIncrement();
+    cluster.getConnector("root", "secret").tableOperations().create(exportTable);
+
+    PrintProps.configureApplication(config,
+        new TableInfo(cluster.getInstanceName(), cluster.getZooKeepers(), "root", "secret",
+            exportTable), 5);
+
+    FluoFactory.newAdmin(config).initialize(
+        new FluoAdmin.InitOpts().setClearTable(true).setClearZookeeper(true));
+
+    miniFluo = FluoFactory.newMiniFluo(config);
+  }
+
+
+  @After
+  public void tearDownFluo() throws Exception {
+    if (miniFluo != null) {
+      miniFluo.close();
+    }
+  }
+
+  public static List<Page> readPages(File input) throws Exception {
+    List<Page> pageList = new ArrayList<>();
+    ArchiveReader ar = WARCReaderFactory.get(input);
+    for (ArchiveRecord r : ar) {
+      Page p = ArchiveUtil.buildPage(r);
+      if (p.isEmpty() || p.getOutboundLinks().isEmpty()) {
+        continue;
+      }
+      pageList.add(p);
+    }
+    ar.close();
+    return pageList;
+  }
+
+  @Test
+  public void testIndexing() throws Exception {
+
+    List<Page> pages = readPages(new File("src/test/resources/wat-18.warc"));
+
+    // First use Spark to create index
+    JavaRDD<Page> pagesRDD = sc.parallelize(pages);
+    IndexStats stats = new IndexStats(sc);
+    JavaPairRDD<RowColumn, Bytes> linkIndex = IndexUtil.createLinkIndex(stats, pagesRDD);
+    List<Tuple2<RowColumn, Bytes>> linkIndexList = linkIndex.collect();
+
+    JavaPairRDD<RowColumn, Bytes> linkIndexNoRank = IndexUtil.filterRank(linkIndex);
+
+    Assert.assertEquals(4, pagesRDD.count());
+    Assert.assertEquals(126, linkIndex.count());
+
+    // Next use Fluo to create index
+    try (FluoClient client = FluoFactory.newClient(miniFluo.getClientConfiguration())) {
+
+      try (LoaderExecutor le = client.newLoaderExecutor()) {
+        for (Page page : pages) {
+          log.debug("Loading page {} with {} links", page.getUrl(), page.getOutboundLinks().size());
+          le.execute(PageUpdate.updatePage(page));
+        }
+      }
+
+      miniFluo.waitForObservers();
+
+      boolean foundDiff = false;
+      foundDiff |= diffExportTable(linkIndexList);
+      foundDiff |= diffFluoTable(client, linkIndexNoRank.collect());
+      if (foundDiff) {
+        dump(client);
+        dumpExportTable();
+        print(linkIndexList);
+      }
+      Assert.assertFalse(foundDiff);
+
+      String url = "http://1000games.me/games/gametion/";
+      log.info("Deleting page {}", url);
+      try (LoaderExecutor le = client.newLoaderExecutor()) {
+        le.execute(PageUpdate.deletePage(url));
+      }
+
+      miniFluo.waitForObservers();
```

Comment:
should there be some sort of verification that the correct updates were made for deleting this page?